src/sqlfluff/core/parser/segments/base.py

```diff
@@ -391,6 +423,38 @@ def raw(self) -> str:
         """Make a string from the segments of this segment."""
         return "".join(seg.raw for seg in self.segments)
 
+    @cached_property
+    def class_types(self) -> Set[str]:
+        """The set of types for this segment."""
+        # NOTE: This version is simple, but some dependent classes
+        # (notably RawSegment) override this with something more
+        # custom.
+        return self._class_types
+
+    @cached_property
+    def child_type_set(self) -> Set[str]:
```

Comment:
Would it be clearer to call this `descendant_type_set` (child, grandchild, etc.)?

Reply:
Good idea.